File: public/generate-icons.py
#!/usr/bin/env python3
"""
Gerador de Ícones PWA
Gera todos os ícones necessários para o PWA
"""

try:
    from PIL import Image, ImageDraw, ImageFont
    import os
except ImportError:
    print("❌ Erro: Biblioteca PIL/Pillow não instalada.")
    print("\nPara instalar:")
    print("  pip install Pillow")
    print("\nOu use o gerador HTML: public/images/icon-generator.html")
    exit(1)

# Tamanhos dos ícones
sizes = [32, 72, 96, 128, 144, 152, 192, 384, 512]
output_dir = os.path.join(os.path.dirname(__file__), 'images')

# Criar diretório se não existir
os.makedirs(output_dir, exist_ok=True)

def generate_icon(size):
    """Gera um ícone do tamanho especificado"""
    # Criar imagem com fundo transparente
    img = Image.new('RGBA', (size, size), (0, 0, 0, 0))
    draw = ImageDraw.Draw(img)
    
    # Cores
    bg_dark_1 = (31, 41, 55)  # #1f2937
    bg_dark_2 = (55, 65, 81)  # #374151
    primary = (16, 185, 129)  # #10b981
    primary_dark = (5, 150, 105)  # #059669
    primary_light = (52, 211, 153)  # #34d399
    primary_darker = (4, 120, 87)  # #047857
    
    # Fundo gradiente (simulado com linhas)
    for y in range(size):
        ratio = y / size
        r = int(bg_dark_1[0] + ratio * (bg_dark_2[0] - bg_dark_1[0]))
        g = int(bg_dark_1[1] + ratio * (bg_dark_2[1] - bg_dark_1[1]))
        b = int(bg_dark_1[2] + ratio * (bg_dark_2[2] - bg_dark_1[2]))
        draw.line([(0, y), (size, y)], fill=(r, g, b))
    
    # Tamanhos para o ícone de caixa
    center_x = size / 2
    center_y = size / 2
    icon_size = size * 0.5
    box_size = icon_size * 0.6
    box_x = center_x - box_size / 2
    box_y = center_y - box_size / 2
    offset = box_size * 0.15
    
    # Sombra
    shadow_box = (
        int(box_x + offset),
        int(box_y + offset),
        int(box_x + offset + box_size),
        int(box_y + offset + box_size)
    )
    draw.rectangle(shadow_box, fill=(0, 0, 0, 76))  # rgba(0,0,0,0.3)
    
    # Face frontal (caixa)
    front_box = (
        int(box_x),
        int(box_y),
        int(box_x + box_size),
        int(box_y + box_size)
    )
    draw.rectangle(front_box, fill=primary, outline=primary_dark, width=max(1, int(size * 0.015)))
    
    # Face superior
    top_points = [
        (int(box_x), int(box_y)),
        (int(box_x + offset), int(box_y - offset)),
        (int(box_x + offset + box_size), int(box_y - offset)),
        (int(box_x + box_size), int(box_y))
    ]
    draw.polygon(top_points, fill=primary_light, outline=primary_dark)
    
    # Face lateral
    side_points = [
        (int(box_x + box_size), int(box_y)),
        (int(box_x + box_size + offset), int(box_y - offset)),
        (int(box_x + box_size + offset), int(box_y - offset + box_size)),
        (int(box_x + box_size), int(box_y + box_size))
    ]
    draw.polygon(side_points, fill=primary_dark, outline=primary_darker)
    
    # Linhas divisórias
    line_width = max(1, int(size * 0.01))
    
    # Linha horizontal superior
    draw.line([
        (int(box_x + box_size * 0.5), int(box_y)),
        (int(box_x + box_size * 0.5 + offset), int(box_y - offset))
    ], fill=primary_darker, width=line_width)
    
    # Linha horizontal inferior
    draw.line([
        (int(box_x + box_size * 0.5), int(box_y + box_size)),
        (int(box_x + box_size * 0.5 + offset), int(box_y - offset + box_size))
    ], fill=primary_darker, width=line_width)
    
    # Linha vertical divisória
    draw.line([
        (int(box_x + box_size * 0.5), int(box_y)),
        (int(box_x + box_size * 0.5), int(box_y + box_size))
    ], fill=primary_darker, width=line_width)
    
    return img

# Gerar todos os ícones
print("🎨 Gerando ícones PWA...")
print()

generated = []

for size in sizes:
    print(f"Gerando ícone {size}x{size}...", end=" ")
    img = generate_icon(size)
    
    if size == 32:
        filename = os.path.join(output_dir, 'favicon.png')
    else:
        filename = os.path.join(output_dir, f'icon-{size}x{size}.png')
    
    img.save(filename, 'PNG', optimize=True)
    generated.append(os.path.basename(filename))
    print("✅")

print()
print("✅ Ícones gerados com sucesso!")
print()
print("Arquivos criados em public/images/:")
for filename in generated:
    print(f"  ✓ {filename}")

File: public/generate-icons.php
<?php
/**
 * Gerador de Ícones PWA
 * 
 * NOTA: Este script requer a extensão GD do PHP habilitada.
 * Se não tiver GD, use o gerador HTML: public/images/icon-generator.html
 * 
 * Para habilitar GD no PHP:
 * - Windows: Descomente extension=gd no php.ini
 * - Linux: sudo apt-get install php-gd
 */

// Verificar se GD está disponível
if (!extension_loaded('gd')) {
    die("❌ Erro: Extensão GD não está habilitada no PHP.\n\n" .
        "Por favor, use o gerador HTML alternativo:\n" .
        "1. Abra: public/images/icon-generator.html no navegador\n" .
        "2. Clique em 'Gerar Todos os Ícones'\n" .
        "3. Os ícones serão baixados automaticamente\n" .
        "4. Coloque os arquivos na pasta public/images/\n\n" .
        "Ou habilite a extensão GD no PHP.\n");
}

// Tamanhos dos ícones
$sizes = [72, 96, 128, 144, 152, 192, 384, 512];
$outputDir = __DIR__ . '/images/';

// Criar diretório se não existir
if (!is_dir($outputDir)) {
    mkdir($outputDir, 0755, true);
}

function generateIcon($size) {
    // Criar imagem
    $img = imagecreatetruecolor($size, $size);
    
    // Cores
    $bgDark1 = imagecolorallocate($img, 31, 41, 55);  // #1f2937
    $bgDark2 = imagecolorallocate($img, 55, 65, 81);  // #374151
    $primary = imagecolorallocate($img, 16, 185, 129); // #10b981
    $primaryDark = imagecolorallocate($img, 5, 150, 105); // #059669
    $primaryLight = imagecolorallocate($img, 52, 211, 153); // #34d399
    $primaryDarker = imagecolorallocate($img, 4, 120, 87); // #047857
    $white = imagecolorallocate($img, 255, 255, 255);
    $shadow = imagecolorallocatealpha($img, 0, 0, 0, 76); // rgba(0,0,0,0.3)
    
    // Fundo gradiente (simulado)
    for ($y = 0; $y < $size; $y++) {
        $ratio = $y / $size;
        $r = (int)(31 + ($ratio * (55 - 31)));
        $g = (int)(41 + ($ratio * (65 - 41)));
        $b = (int)(55 + ($ratio * (81 - 55)));
        $color = imagecolorallocate($img, $r, $g, $b);
        imageline($img, 0, $y, $size, $y, $color);
    }
    
    // Tamanhos para o ícone
    $centerX = $size / 2;
    $centerY = $size / 2;
    $iconSize = $size * 0.5;
    $boxSize = $iconSize * 0.6;
    $boxX = $centerX - $boxSize / 2;
    $boxY = $centerY - $boxSize / 2;
    $offset = $boxSize * 0.15;
    
    // Sombra
    imagefilledrectangle($img, 
        (int)($boxX + $offset), 
        (int)($boxY + $offset), 
        (int)($boxX + $offset + $boxSize), 
        (int)($boxY + $offset + $boxSize), 
        $shadow
    );
    
    // Face frontal (caixa)
    imagefilledrectangle($img, 
        (int)$boxX, 
        (int)$boxY, 
        (int)($boxX + $boxSize), 
        (int)($boxY + $boxSize), 
        $primary
    );
    
    // Borda da face frontal
    imagerectangle($img, 
        (int)$boxX, 
        (int)$boxY, 
        (int)($boxX + $boxSize), 
        (int)($boxY + $boxSize), 
        $primaryDark
    );
    
    // Face superior
    $points = [
        (int)$boxX, (int)$boxY,
        (int)($boxX + $offset), (int)($boxY - $offset),
        (int)($boxX + $offset + $boxSize), (int)($boxY - $offset),
        (int)($boxX + $boxSize), (int)$boxY
    ];
    imagefilledpolygon($img, $points, 4, $primaryLight);
    imagepolygon($img, $points, 4, $primaryDark);
    
    // Face lateral
    $points = [
        (int)($boxX + $boxSize), (int)$boxY,
        (int)($boxX + $boxSize + $offset), (int)($boxY - $offset),
        (int)($boxX + $boxSize + $offset), (int)($boxY - $offset + $boxSize),
        (int)($boxX + $boxSize), (int)($boxY + $boxSize)
    ];
    imagefilledpolygon($img, $points, 4, $primaryDark);
    imagepolygon($img, $points, 4, $primaryDarker);
    
    // Linhas divisórias
    imageline($img, 
        (int)($boxX + $boxSize * 0.5), 
        (int)$boxY, 
        (int)($boxX + $boxSize * 0.5 + $offset), 
        (int)($boxY - $offset), 
        $primaryDarker
    );
    
    imageline($img, 
        (int)($boxX + $boxSize * 0.5), 
        (int)($boxY + $boxSize), 
        (int)($boxX + $boxSize * 0.5 + $offset), 
        (int)($boxY - $offset + $boxSize), 
        $primaryDarker
    );
    
    // Linha vertical divisória
    imageline($img, 
        (int)($boxX + $boxSize * 0.5), 
        (int)$boxY, 
        (int)($boxX + $boxSize * 0.5), 
        (int)($boxY + $boxSize), 
        $primaryDarker
    );
    
    return $img;
}

// Gerar todos os ícones
echo "Gerando ícones PWA...\n";
$generated = [];

foreach ($sizes as $size) {
    echo "Gerando ícone {$size}x{$size}...\n";
    $img = generateIcon($size);
    $filename = $outputDir . "icon-{$size}x{$size}.png";
    imagepng($img, $filename);
    imagedestroy($img);
    $generated[] = "icon-{$size}x{$size}.png";
}

// Gerar favicon também (32x32)
echo "Gerando favicon...\n";
$favicon = generateIcon(32);
imagepng($favicon, $outputDir . 'favicon.png');
imagedestroy($favicon);
$generated[] = 'favicon.png';

echo "\n✅ Ícones gerados com sucesso!\n";
echo "Arquivos criados:\n";
foreach ($generated as $file) {
    echo "  - images/{$file}\n";
}

